src/hotspot/share/opto/subnode.cpp

```diff
@@ -1529,10 +1529,46 @@ Node *BoolNode::Ideal(PhaseGVN *phase, bool can_reshape) {
     }
   }
 
+  const int cmp1_op = cmp1->Opcode();
+  const int cmp2_op = cmp2->Opcode();
+
+  // Change x +- Integer.MIN_VALUE <=> y +- Integer.MIN_VALUE into x u<=> y
+  if ((_test._test == BoolTest::lt || _test._test == BoolTest::le ||
+        _test._test == BoolTest::gt || _test._test == BoolTest::ge) &&
```

Comment:
You can use `(_test. is_less() || _test.is_greater())` here.

Reply:
Thank you very much for the review, I have addressed this and the following comment.